### PR TITLE
[Dashboards] Add new properties to group widget, note widget and image widget

### PR DESCRIPTION
**Pull request by DavidRobertAnsart**
### Changes
With the multi-size layout, note widget and image widget have been updated with new properties:

- Replace 'multi-size layout' with 'new dashboard layout' in the reflow type description.
- Add `is_column_break` to widget layouts. (Optional property for multi-size layout only.)
- Add `url_dark_theme`, `has_background` , `has_border`, `horizontal_align` and `vertical_align` to image widget.
- Add `vertical_align` and `has_padding` to note widget.
- Add `background_color`, `show_title` and `banner_img` to group widget.

**Documentation PR:** https://github.com/DataDog/datadog-api-spec/pull/944 & DataDog/datadog-api-spec#912

### CODE REVIEW

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -1174,6 +1188,22 @@ func getGroupDefinitionSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Description: "The title of the group.",
 		},
+		"background_color": {
```
Comment (matt-miller-ddog) [inline datadog/resource_datadog_dashboard.go:1191]: The PR description didn't mention this field; is this intended to be on this PR?

File: datadog/resource_datadog_dashboard.go [hunk 2]
```diff
@@ -1174,6 +1188,22 @@ func getGroupDefinitionSchema() map[string]*schema.Schema {
...
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Background color of the group title.",
```
Comment (matt-miller-ddog) [inline datadog/resource_datadog_dashboard.go:1194]: What kinds of values are allowed for this field?

File: datadog/resource_datadog_dashboard.go (hunk 1 above)
Comment (DavidRobertAnsart) [reply datadog/resource_datadog_dashboard.go:1191]: Yes it is, I just forgot it. It was not added in DataDog/datadog-api-spec#944 but already before here: https://github.com/DataDog/datadog-api-spec/pull/912

File: datadog/resource_datadog_dashboard.go (hunk 2 above)
Comment (DavidRobertAnsart) [reply datadog/resource_datadog_dashboard.go]: The background_color field accepts any string, but only these values will render a colored header: `vivid_blue`, `vivid_purple`, `vivid_pink`, `vivid_orange`, `vivid_yellow`, `vivid_green`, `blue`, `purple`, `pink`, `orange`, `yellow`, `green`, `gray` or `white`.
-> I updated the description with the values.

**Review by matt-miller-ddog — DISMISSED** [review]

**Review by therve — APPROVED** [review]

**Review by matt-miller-ddog — APPROVED** [review]

**Pull request merged**